qiskit_ibm_runtime/options/measure_noise_learning_options.py

```diff
@@ -0,0 +1,41 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Options for measurement noise learning."""
+
+from typing import Union
+
+from .utils import Unset, UnsetType, primitive_dataclass, make_constraint_validator
+
+
+@primitive_dataclass
+class MeasureNoiseLearningOptions:
+    """Options for measurement noise learning.
+
+    .. note::
+        These options are only used when the resilience level or options specify a
+        technique that requires measurement noise learning.
+
+    Args:
+        meas_num_randomizations: The number of random circuits to draw for the measurement
+            learning experiment. Default: 32.
+
+        meas_shots_per_randomization: The number of shots to use for the learning experiment
```

Comment:
The `"auto"` is the default value here, we should also document the default heuristic which scales the total number of shots with the pub precision and number of basis observable being measureed

Reply:
`"auto"` is essentially saying "use whatever the server decides", which is the same as `Unset`.

Reply:
Unset might mean let server decide, but `Literal["auto"]` is only an allowed value for certain options (like this one), and if that is the server default it means the server will choose a value based on some heuristics, not just the server will choose a fixed value.

In absence of having an exact description of the heuristic here, which would probably be better in API docs like you say, we could just say something like: if `"auto"` this value will be chosen automatically basedo n the input pubs (Default `"auto"`), or something along those lines

Reply:
Off the top of my head, all of the places where we have "auto", the default value is "auto". To me, this sort of means that Unset is fine until and if that changes. Doesn't mean we can't describe it. In some places, we do distinguish between None and "auto", where "auto" means "it will have a value, but we'll choose it for you later" and None means "this thing won't ever have a value".

Reply:
Updated in [db05269](https://github.com/Qiskit/qiskit-ibm-runtime/pull/1460/commits/db052691a0591c4cff71c6c7156e4b5c9f7dad1b)